plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -839,7 +840,12 @@ private int findRouterEthDeviceIndex(String domrName, String routerIp, String ma
         // when we dynamically plug in a new NIC into virtual router, it may take time to show up in guest OS
         // we use a waiting loop here as a workaround to synchronize activities in systems
         long startTick = System.currentTimeMillis();
-        while (System.currentTimeMillis() - startTick < 15000) {
+        long waitTimeoutMillis = 15000;
+        Long waitTimeoutMillisLong = VmwareManagerImpl.s_vmwareNicHotplugWaitTimeout.value();
+        if (waitTimeoutMillisLong != null) {
+            waitTimeoutMillis = waitTimeoutMillisLong;
+        }
+        while (System.currentTimeMillis() - startTick < waitTimeoutMillis) {
```

Comment:
The config value can be directly read using s_vmwareNicHotplugWaitTimeout.value(), no need for the checks.